src/app/mapControls/Filter.js

```diff
@@ -0,0 +1,227 @@
+                    show: 750,
+                    hide: 100
+                },
+                container: 'body',
+                html: true,
+                placement: 'bottom'
+            });
+            this.inherited(arguments);
+        },
+        clear: function () {
+            // summary:
+            //      unselects all buttons
+            console.log('app/mapControls/Filter:clear', arguments);
+
+            query('.btn', this.buttonContainer).forEach(function (btn) {
+                domClass.remove(btn, 'active');
+                btn.setAttribute('aria-pressed', false);
+            });
+
+            this.numSpan.innerHTML = 0;
+            this.selectedValues = [];
+            this.emit('changed');
+        },
+        itemClicked: function (value) {
+            // summary:
+            //      description
+            // value: String
+            //      value of the item that was clicked
+            console.log('app/mapControls/Filter:itemClicked', arguments);
+
+            var index = this.selectedValues.indexOf(value);
+            if (index === -1) {
+                this.selectedValues.push(value);
+            } else {
+                this.selectedValues.splice(index, 1);
+            }
+
+            this.numSpan.innerHTML = this.selectedValues.length;
+
+            this.emit('changed');
+        },
+        getQuery: function () {
+            // summary:
+            //      assembles all selected values into a def query
+            console.log('app/mapControls/Filter:getQuery', arguments);
+
+            if (this.selectedValues.length) {
+                var values;
+                if (this.fieldType === c.TYPE_TEXT) {
+                    values = this.selectedValues.map(function (v) {
+                        return "'" + v + "'";
+                    });
+                } else {
+                    values = this.selectedValues;
+                }
+                if (this.any) {
+                    var where = this.fieldName + ' IN (' + values.join(', ') + ')';
+                    return this.getRelatedTableQuery(where);
+                } else {
+                    var that = this;
+                    return values.reduce(function (previousReturn, currentValue) {
+                        var where = that.fieldName + ' = ' + currentValue;
+                        if (!previousReturn) {
+                            return that.getRelatedTableQuery(where);
+                        } else {
+                            return previousReturn + ' AND ' + that.getRelatedTableQuery(where);
+                        }
+                    }, false);
+                }
+            } else {
+                return undefined;
+            }
+        },
+        toggleAny: function () {
+            // summary:
+            //      description
+            console.log('app/mapControls/Filter:toggleAny', arguments);
+
+            var that = this;
+            setTimeout(function () {
```

Comment:
To bump that code to the bottom of the event loop. This gives bootstrap a chance to apply the `active` label to the button that the code checks for. I've added clarifying comments to this code.